aten/src/ATen/native/cpu/MaxPoolKernel.cpp

```diff
@@ -85,15 +85,22 @@ compute_internal(
   using iVec = vec::Vectorized<integer_t>;
   // Pass I: init out lane
   iVec index0_vec = iVec(id0 * input_height * input_width + ih0 * input_width + iw0);
-  Vec out_vec = Vec(-std::numeric_limits<scalar_t>::infinity());
+
+  scalar_t min_value;
+  if (std::is_integral<scalar_t>::value) {
+    min_value = std::numeric_limits<scalar_t>::min();
```

Comment:
NIT: In `ATen/native/ReduceOpsUtils.h` there is a helper `lower_bound` which does exactly what you want here.